packages/header-checker-lint/src/header-checker-lint.ts

```diff
@@ -39,15 +40,72 @@ interface LicenseHeader {
   year?: number;
 }
 
-const SOURCE_FILE_TYPES = ['ts', 'js', 'java'];
+interface ConfigurationOptions {
+  allowedCopyrightHolders: string[];
+  allowedLicenses: LicenseType[];
+  ignoreFiles: string[];
+  sourceFileExtensions: string[];
+}
+
+const WELL_KNOWN_CONFIGURATION_FILE = '.bots/header-checker-lint.json';
+const DEFAULT_CONFIGURATION: ConfigurationOptions = {
+  allowedCopyrightHolders: ['Google LLC'],
+  allowedLicenses: ['Apache-2.0', 'MIT'],
+  ignoreFiles: [],
+  sourceFileExtensions: ['ts', 'js', 'java'],
+};
 
-function isSourceFile(file: string): boolean {
-  const extension = file.substring(file.lastIndexOf('.') + 1);
-  return SOURCE_FILE_TYPES.includes(extension);
+class Configuration {
+  private options: ConfigurationOptions;
+
+  constructor(options: ConfigurationOptions) {
+    this.options = options;
+  }
+
+  static async fromGitHub(
+    path: string,
+    owner: string,
+    repo: string,
+    ref: string,
+    github: GitHubAPI
+  ): Promise<Configuration> {
+    return github.repos
```

Comment:
no need for the promise chain here, in an async function, you can just:

```
try {
await response = github.repos.getContents();
} catch () {}
```

Reply:
done